dbms/src/Server/BgStorageInit.cpp

```diff
@@ -93,7 +71,8 @@ void BgStorageInitHolder::start(Context & global_context, const LoggerPtr & log,
             }
         };
 
-        setStorageInitThreadPool(global_context);
+        size_t default_num_threads = std::max(4UL, 2 * std::thread::hardware_concurrency()) * global_context.getSettingsRef().io_thread_count_scale * 5 ;
```

Comment:
It's tricky to be `lobal_context.getSettingsRef().io_thread_count_scale * 5`. How about just assigning a new settings for it?